src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -215,34 +210,56 @@ internal static bool TryGetApiVersion(Uri startRequestUri, out ReadOnlySpan<char
         /// <summary>
         /// This function is used to get the final request uri after the lro has completed.
         /// </summary>
-        private string? GetFinalUri()
+        private string? GetFinalUri(Response response)
         {
             // Set final uri as null if the response header doesn't contain "Operation-Location" or "Azure-AsyncOperation".
             if (_headerSource is not (HeaderSource.OperationLocation or HeaderSource.AzureAsyncOperation))
             {
                 return null;
             }
 
-            // Set final uri as null if original request is a delete method.
-            if (_requestMethod == RequestMethod.Delete)
+            if (_finalStateVia == OperationFinalStateVia.AzureAsyncOperation && _requestMethod == RequestMethod.Post)
             {
                 return null;
             }
 
-            // Set final uri as original request's uri if one of these requirements are met:
-            // 1.Original request is a put method;
-            // 2.Original request is a management plane patch method. For data plane patch method, the final uri will be determinned by the original response header and "_finalStateVia";
-            // 3.Original response header contains "Location" and FinalStateVia is configured to OriginalUri.
-            if (_requestMethod == RequestMethod.Put || _requestMethod == RequestMethod.Patch && _startRequestUri.Scheme == "https" && _startRequestUri.Host == "management.azure.com" || _originalResponseHasLocation && _finalStateVia == OperationFinalStateVia.OriginalUri)
+            // If body contains resourceLocation, use it
+            if (response.ContentStream?.Length > 0)
+            {
+                try
+                {
+                    using JsonDocument document = JsonDocument.Parse(response.ContentStream);
+                    var root = document.RootElement;
+                    if (root.TryGetProperty("resourceLocation", out var resourceLocation))
+                    {
+                        return resourceLocation.GetRequiredString();
+                    }
+                }
+                finally
+                {
+                    // It is required to reset the position of the content after reading as this response may be used for deserialization.
+                    response.ContentStream.Position = 0;
```

Comment:
Are we sure that all streams are seekable? We don't expect any `NetworkStream` implementation, for example (which you cannot seek)?

Reply:
Good point, will add a check

Reply:
fixed